cpp/tests/binaryop/binop-fixture.hpp

```diff
@@ -27,36 +27,56 @@ namespace cudf {
 namespace test {
 namespace binop {
 
-struct BinaryOperationTest
-    : public cudf::test::BaseFixture
-    , cudf::test::UniformRandomGenerator<uint8_t>
+struct BinaryOperationTest : public cudf::test::BaseFixture
 {
-    BinaryOperationTest()
-     : cudf::test::UniformRandomGenerator<uint8_t>(1, 10)
-    {}
+    BinaryOperationTest() {}
 
-    auto make_data_iter() {
+    static constexpr int r_min = 1;
+    static constexpr int r_max = 10;
+
+    template<typename T>
+    auto make_data_iter(cudf::test::UniformRandomGenerator<T>& rand_gen) {  
         return cudf::test::make_counting_transform_iterator(0,
-            [this](auto row) { return this->generate(); });
+            [&](auto row) { return rand_gen.generate(); });
     }
 
     auto make_validity_iter() {
-        uint8_t mod_base = this->generate();
+        cudf::test::UniformRandomGenerator<uint8_t> rand_gen(r_min, r_max);
+        uint8_t mod_base = rand_gen.generate();
         return cudf::test::make_counting_transform_iterator(0,
             [mod_base](auto row) { return (row % mod_base) > 0; });
     }
+    
+    template <typename T, typename std::enable_if_t<cudf::is_timestamp<T>()>* = nullptr >
+    auto make_random_wrapped_column(size_type size) {
+        cudf::test::UniformRandomGenerator<long> rand_gen(r_min, r_max);
```

Comment:
`UniformRandomGenerator` should already be specialized to avoid needing to do this kind of extra specialization.

Reply:
This doesn't work because the uniform_distribution field gets defined like this:

```
template<typename T>
struct uniform_distribution_impl<T, std::enable_if_t<cudf::is_timestamp<T>() > > {
   using type = std::uniform_int_distribution<typename T::duration::rep>;
};
```

which turns into

std::uniform_int_distribution<long>

And it won't let you initialize it with timestamps.

Reply:
Then `UniformRandomGenerator` needs to be modified. SFINAE its ctor for timestamp types.

Reply:
Yep. Good call.